driver-core/src/main/com/mongodb/internal/dns/DefaultDnsResolver.java

```
@@ -0,0 +1,148 @@
+    // ':' in between, as expected by ServerAddress
+    // It's required that the srvHost has at least three parts (e.g. foo.bar.baz) and that all of the resolved hosts have a parent
+    // domain equal to the domain of the srvHost.
+    @Override
+    public List<String> resolveHostFromSrvRecords(final String srvHost) {
+        String srvHostDomain = srvHost.substring(srvHost.indexOf('.') + 1);
+        List<String> srvHostDomainParts = asList(srvHostDomain.split("\\."));
+        List<String> hosts = new ArrayList<String>();
+        InitialDirContext dirContext = createDnsDirContext();
+        try {
+            Attributes attributes = dirContext.getAttributes("_mongodb._tcp." + srvHost, new String[]{"SRV"});
+            Attribute attribute = attributes.get("SRV");
+            if (attribute == null) {
+                throw new MongoConfigurationException("No SRV records available for host " + srvHost);
+            }
+            NamingEnumeration<?> srvRecordEnumeration = attribute.getAll();
+            while (srvRecordEnumeration.hasMore()) {
+                String srvRecord = (String) srvRecordEnumeration.next();
+                String[] split = srvRecord.split(" ");
+                String resolvedHost = split[3].endsWith(".") ? split[3].substring(0, split[3].length() - 1) : split[3];
+                String resolvedHostDomain = resolvedHost.substring(resolvedHost.indexOf('.') + 1);
+                if (!sameParentDomain(srvHostDomainParts, resolvedHostDomain)) {
+                    throw new MongoConfigurationException(
+                            format("The SRV host name '%s'resolved to a host '%s 'that is not in a sub-domain of the SRV host.",
+                                    srvHost, resolvedHost));
+                }
+                hosts.add(resolvedHost + ":" + split[2]);
+            }
+
+            if (hosts.isEmpty()) {
```

This is the one change: throw instead of return an empty list.  Makes it a bit easier on the caller

---

Ack